flink-streaming-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -570,139 +563,165 @@ private void closeCurrentPartFile(BucketState<T> bucketState) throws Exception {
 
 	/**
 	 * Gets the truncate() call using reflection.
-	 *
 	 * <p>
-	 * Note: This code comes from Flume
+	 * <b>NOTE:</b> This code comes from Flume.
 	 */
 	private Method reflectTruncate(FileSystem fs) {
-		Method m = null;
```

Comment:
we can heavily reduce the changes made here, by putting
```
if (this.refTruncated != null) {
	return this.refTruncated;
}
```
at the start of the method. the remaining code will remain identical, aside from the new assignment near the end.